### PR TITLE
Feature/user login api

### PR DESCRIPTION
**Pull request by yaman3bd**

### CODE REVIEW

**Review by felmez — APPROVED** [review]
good work

File: src/routers/user.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const express = require('express');
+
+const routes = express.Router();
+
+const { check } = require('express-validator');
+const UserController = require('../controllers/user');
+
+routes.post(
+  'login',
```
Comment (felmez) [inline src/routers/user.js:9]: maybe '/login'

File: src/routers/user.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    .bail(),
+  check('password', 'Password must be at least 5 characters long')
+    .notEmpty()
+    .withMessage('Password should not be empty')
+    .bail(),
+  UserController.login
+);
+
+routes.post(
+  'register',
```
Comment (felmez) [inline src/routers/user.js:22]: here also '/register' since its coming from /users but there it will be /usersregister

**Review by felmez — APPROVED** [review]

File: src/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+const register = async (req, res) => {
+  try {
+    const result = validationResult(req);
+
+    if (!result.isEmpty()) {
+      return res.status(400).json(result.errors);
+    }
+    const { username, password, email, name } = req.body;
+
```
Comment (halitbatur) [inline src/controllers/user.js:57]: extra new line

File: src/controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  }
+};
+
+const register = async (req, res) => {
+  try {
+    const result = validationResult(req);
+
+    if (!result.isEmpty()) {
+      return res.status(400).json(result.errors);
+    }
```
Comment (halitbatur) [inline src/controllers/user.js:55]: new line needed after if

File: src/controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      name,
+      username,
+      password: passwordHashed,
+      email,
+    });
+    return res.status(201).json(newUser);
+  } catch (err) {
+    return res.status(400).send(err);
+  }
+};
```
Comment (halitbatur) [inline src/controllers/user.js:74]: new line needed after this

File: src/controllers/user.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+const logout = async (req, res) => {
```
Comment (halitbatur) [inline src/controllers/user.js:75]: this function is declared async but I cannot see any `await` being used inside of it.

File: src/controllers/user.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  try {
+    res.clearCookie('_t');
+    return res.status(200).json({ success: true });
+  } catch (err) {
+    return res.status(400).send(err);
+  }
+};
```
Comment (halitbatur) [inline src/controllers/user.js:82]: new line needed

File: src/routers/user.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
+const express = require('express');
+
+const routes = express.Router();
+
+const { check } = require('express-validator');
```
Comment (halitbatur) [inline src/routers/user.js:5]: require statements should be at the beginning please move these after requiring express.

File: src/routers/user.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    .isLength({ min: 5 })
+    .bail()
+    .matches(/(?=.*\d)(?=.*[a-z])(?=.*[A-Z])/)
+    .withMessage('Password must contain a number, uppercase and lowercase')
+    .bail()
+    .custom((val, { req }) => val === req.body.confirmPassword)
+    .withMessage('Passwords are not matching')
+    .bail(),
+  UserController.register
+);
```
Comment (halitbatur) [inline src/routers/user.js:53]: new line after this

**Review by halitbatur — CHANGES_REQUESTED** [review]
Overall good job, some a few changes that needs to be addressed.

**Review by halitbatur — APPROVED** [review]
@yaman3bd good work!

**Comment by halitbatur** [comment]
@yaman3bd update branch and solve conflicts then its ready to be merged.

**Pull request merged**